src/compiler/cpp_plugin.cc

```diff
@@ -58,18 +58,35 @@ class CppGrpcGenerator : public grpc::protobuf::compiler::CodeGenerator {
       return false;
     }
 
+    grpc_cpp_generator::Parameters generator_parameters;
+
+    if (!parameter.empty()) {
+      std::vector<grpc::string> parameters_list =
+        grpc_generator::tokenize(parameter, ",");
+      for (auto &parameter_string: parameters_list) {
```

Comment:
Vijay seems to be removing all range based loops, does it apply to the code generators?